File: String/Palindrome-String/PalindromeString.java
public class PalindromeString {

    public static boolean isPalindrome(String str) {
        if (str == null || str.length() == 0) {
            return true;
        }
        int left = 0;
        int right = str.length() - 1;

        while (left < right) {
            if (str.charAt(left) != str.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }

    public static void main(String[] args) {
        String testString = "racecar";
        System.out.println("Is \"" + testString + "\" a palindrome? " + isPalindrome(testString));
    }
}

File: String/Palindrome-String/PalindromeString.js
function CheckPalindrome(str){
    for (let i = 0; i < str.length / 2; i++) {
        if(str[i] !== str[str.length - 1 - i]) {
            console.log("Not a Palindrome");
            return;
        }
    }
    console.log("Is a Palindrome");
}

let str = 'aabbaaa'
CheckPalindrome(str);